### DISCUSSION TITLE
CA Chain Files are not included in the signed PDF

### DISCUSSION THREAD

**Discussion by luiz-quintino** [MatthiasValvekens/pyHanko#255] [Q&A]
The parameter `ca_chain_files=`, from `signers.SimpleSigner.load()` is used to add in the signed PDF the list of certificate atuthority till the root. Then you can see the **chain of thrust** when you are verifying the signature locally.

`signer = signers.SimpleSigner.load(key_file=key_file,
key_passphrase=password, cert_file=cert_file,  ca_chain_files=['cert/root.pem', 'cert/Intermediate.pem'])`

This parameter is not working. The only visible certificate in the signed PDF document is the `cert_file=`, the certificate used to provide the signature. The cert chain is not visible in the document.

Any idea about this issue???

**Comment by MatthiasValvekens** [comment]
Can you elaborate on what you mean by "the cert chain is not visible in the document"? Does your viewer trust your root cert?


If you're using Acrobat: whether you embed the root or not doesn't matter. Either way, Acrobat will only display certification paths to roots in its own trust store (pretty reasonable, really).

The intermediate cert is a different story, but let's validate those assumptions first before trying to debug things. :)

It would also help if you could share some output.

**Comment by luiz-quintino** [comment]
Hi Matthias.

**I made some tests here with this result:**
I opened the signed document in the PDFXChange and Acrobat Viewer. When you clink in the signaure and select to see the certificate this is the result:
- In the Acrobat I can see the certificate chain (signer, intermediates and root).
- I the XChange, I can see see only the signer certificate.

After reeinstal in the windows the signer.pfx file with the entire root I can see also in the XChange the complete tree.
Seems that the Acrobat has an embeded viewer for the certificate and the XChange uses the windows viewer.

Thank you for your support!

**Reply by MatthiasValvekens** [reply]
Alright, so I understand that this is an issue with PDFXChange, and not with the produced output, then? :)
